libqtile/widget/tasklist.py

```diff
@@ -137,16 +185,29 @@ def get_taskname(self, window):
         appropriate characters are prepended.
         """
         state = ''
+        markup_str = self.markup_normal
         if window is None:
             pass
         elif window.minimized:
             state = self.txt_minimized
+            markup_str = self.markup_minimized
         elif window.maximized:
             state = self.txt_maximized
+            markup_str = self.markup_maximized
         elif window.floating:
             state = self.txt_floating
+            markup_str = self.markup_floating
+        elif window is window.group.currentWindow:
+            markup_str = self.markup_focused
+
+        window_name = window.name if window and window.name else "?"
+
+        if markup_str is not None:
+            if self.markup:
+                window_name = pangocffi.markup_escape_text(window_name)
+            return markup_str.format(window_name)
```

Comment:
So it would be nice if we didn't have to add a whole new lot of parameters to control whether or not `pangocffi.markup_escape_text` is applied to some text or not, since text without formatting will work fine here. I think it would be nice if we could just add the `markup_escape_text` call, but I see since we need to `format` the window name between the formatting tags, we can't make the API quite the same. I would say, always `markup_escape_text`, then have a flag that determines whether the returned text is `state.format(window_name)` or `%s%s` % (state, window_name)`. Since this is a new feature, we can require that to get formatting all of the text types to be specified with the format syntax. What do you think about setting it up to work like that?

Reply:
I did rearrange the code, so now there are just two options: use old or new formatting (mixed variants are excluded).